tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -301,12 +301,17 @@ def teardown(self):
         self.webdriver = None
 
     def is_alive(self):
+        webdriver_is_alive = True
+        socket_previous_timeout = socket.getdefaulttimeout()
         try:
-            # Get a simple property over the connection
+            # Get a simple property over the connection, with 2 seconds of timeout
+            socket.setdefaulttimeout(2)
             self.webdriver.window_handle
-        except (socket.timeout, client.UnknownErrorException):
-            return False
-        return True
+        except (socket.timeout, client.UnknownErrorException, client.InvalidSessionIdException):
```

Comment:
You could optionally do this without an extra variable, just `return True` at the end of the `try` block and `return False` at the end of the `except` block. But only if you think it reads better.

Reply:
So i have just discovered that finally clauses execute even if you return from inside the try block :) Interesting. I will change the code to avoid this extra variable as you suggest

Reply:
I just pushed a new amended commit with this changes